### PR TITLE
Upgrade everything to 1.10

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

**Comment by avik-pal** [comment]
@ChrisRackauckas is there a reason SymbolicIndexingInterface is restricted to 1.10 only? I can't install SciMLBase with my patch on 1.9 and I can get the benchmarks to run on 1.10

File: .buildkite/run_benchmark.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ steps:
             - BUILDKITE_S3_SECRET_ACCESS_KEY="U2FsdGVkX1+LWh1yX7LsMBlecEJLc08eJrgOhurhd47CY1/jS3wCGVCQmS1t6f2j70spBoFdfc9kn2naj8HH5A==" 
             - BUILDKITE_S3_DEFAULT_REGION="U2FsdGVkX18ccoE9FmtkwsCm1x0MLMBlN/FLcAyKkY4="
       - JuliaCI/julia#v1:
-          version: 1.9
+          version: 1.10
```
Comment (thazhemadam) [inline .buildkite/run_benchmark.yml:18]: This might need to be update to `version: "1.10"`.

Comment (avik-pal) [reply .buildkite/run_benchmark.yml]: ah that does it. Thanks!

**Comment by avik-pal** [comment]
@thazhemadam can you help me with the "Pipeline '.buildkite/launch_benchmarks.yml' fails treehash signature check! You may need to re-run cryptic/bin/sign_treehashes!". Not sure how to do that.

**Comment by avik-pal** [comment]
Hopefully will be ready to merge after all of these finish running in a few days :stuck_out_tongue_closed_eyes:

**Comment by thazhemadam** [comment]
> @thazhemadam can you help me with the "Pipeline '.buildkite/launch_benchmarks.yml' fails treehash signature check! You may need to re-run cryptic/bin/sign_treehashes!". Not sure how to do that.

Done.

**Comment by ChrisRackauckas** [comment]
Updating everything to v1.10 all at once seems like a scary idea. I recommend just doing a folder at a time.

**Comment by avik-pal** [comment]
Yeah seems like quite a few failures. What I will do is create a list in that issue with all the builds and mark the ones which fail.

**Pull request closed without merging**